api/qan/collector.proto

```diff
@@ -0,0 +1,203 @@
+syntax = "proto3";
+
+package qan;
+
+option go_package = "qan";
+
+// Agent service contains RPCs to implement communication mechanism with qan-agent.
+// Namely query metrics, query explain etc.
+service Agent {
+  // DataInterchange defines duplex communication between Query analytics API and Agent.
+  rpc DataInterchange(stream AgentMessage) returns (stream ApiMessage);
+}
+
+// AgentMessage contains several Query Classes for one or more periods of time.
+message AgentMessage {
+  repeated QueryClass query_class = 1;
+}
+
+// QueryClass is aggregated message created by qan-agent.
+// Contains information about one query selected in defined way from query class in specific period of time.
+message QueryClass {
+  string queryid = 1; // md5 of digest_text/fingerprint.
+  string fingerprint = 2; // digest_text - query signature. Query without values.
+  // Dimension Group.
+  string d_server = 3;
+  string d_database = 4; // postgress database.
+  string d_schema = 5; // mysql database; postgress schema; mongo collection.
+  string d_username = 6; // client user name.
+  string d_client_host = 7; // client IP or hostname.
+  map<string, string> labels = 8; // Custom labels names:values.
+  string agent_uuid = 9; // Identifier of agent that collect and send metrics.
+  int64 period_start = 10; // Time when collection of bucket started.
+  uint32 period_length = 11; // Duration of bucket.
+  string metrics_source = 12; // Ex.: slowlog, perf schema, Mongo DB, Percona Server, etc.
+  string example = 13; // One of query example from set found in bucket.
+  // ExampleFormat is formant of query example: real or query without values.
+  enum ExampleFormat {
+    QUERY_CLASS_EXAMPLE_FORMAT_INVALID = 0;
+    EXAMPLE = 1;
+    DIGEST = 2;
+  }
+  ExampleFormat example_format = 14;
+  uint32 is_truncated = 15; // Indicates if query examples is too long and was truncated.
+  // ExampleType is a type of query example selected for this query class in given period of time.
+  enum ExampleType {
```

Comment:
Duplicate of https://github.com/percona/pmm/pull/268/files#diff-283a55e2672ed0140ef30d059c3be9e0R45.

Reply:
Moved to qan.proto